### PR TITLE
Change huge page assignment automatically to adapt for aarch64.

### PR DESCRIPTION
**Pull request by liang-cong-red-hat**
Test result on aarch64 64k:
 (1/1) type_specific.io-github-autotest-libvirt.memory.devices.lifecycle.mutiple_mem: PASS (147.81 s)

Test result on x86_64:
(1/1) type_specific.io-github-autotest-libvirt.memory.devices.lifecycle.mutiple_mem: PASS (155.30 s)

### CODE REVIEW

File: libvirt/tests/src/memory/memory_devices/lifecycle_of_mixed_memory_devices.py [hunk 1]
```diff
@@ -293,10 +310,12 @@ def setup_test():
         """
         test.log.info("TEST_SETUP: Set hugepage and guest boot ")
         process.run(truncate_cmd, ignore_status=False, shell=True)
-        utils_memory.set_num_huge_pages(int(nr_hugepages))
+        nr_hugepages = math.ceil(
+            int(total_hugepage_size) / int(utils_memory.get_huge_page_size()))
```
Comment (dzhengfy) [inline libvirt/tests/src/memory/memory_devices/lifecycle_of_mixed_memory_devices.py:314]: `utils_memory.get_huge_page_size()` directly return int value

File: libvirt/tests/src/memory/memory_devices/lifecycle_of_mixed_memory_devices.py [hunk 2]
```diff
@@ -293,10 +310,12 @@ def setup_test():
...
+        utils_memory.set_num_huge_pages(nr_hugepages)
         result = process.run("cat %s" % kernel_hp_file, ignore_status=True,
                              verbose=False).stdout_text.strip()
-        if nr_hugepages != result:
+        if str(nr_hugepages) != result:
             test.fail("Hugepage set failed, should be %s instead of %s" % (
                 nr_hugepages, result))
```
Comment (dzhengfy) [inline libvirt/tests/src/memory/memory_devices/lifecycle_of_mixed_memory_devices.py:320]: For line 315 to 320, I suggest next time we use another way like below:

hpc = test_setup.HugePageConfig(params)
hpc.set_kernel_hugepages(pagesize, pagenum)

We need to enhance set_kernel_hugepages method by adding 318-320 checking

File: libvirt/tests/src/memory/memory_devices/lifecycle_of_mixed_memory_devices.py (hunk 1 above)
Comment (liang-cong-red-hat) [reply libvirt/tests/src/memory/memory_devices/lifecycle_of_mixed_memory_devices.py]: done

File: libvirt/tests/src/memory/memory_devices/lifecycle_of_mixed_memory_devices.py (hunk 2 above)
Comment (liang-cong-red-hat) [reply libvirt/tests/src/memory/memory_devices/lifecycle_of_mixed_memory_devices.py:320]: got it

**Review by dzhengfy — APPROVED** [review]
LGTM

**Comment by dzhengfy** [comment]
No many codes logic is changed, instead just few hugepage replacement. So I merge it directly.

**Pull request merged**